### PR TITLE
subsys/testsuite/ztest: relocate `unit_testing` stub headers

### PR DESCRIPTION
**Pull request by ycsin**
These header stubs seems to be required by the `unit_testing` board only, move them into a dedicated folder and include them only when compiling for it.

See also #69461, which generates these headers at build instead of relocating them

### CODE REVIEW

File: include/zephyr/linker/linker-defs.h [hunk 1]
```diff
@@ -25,14 +25,7 @@
 #include <zephyr/sys/util.h>
 #include <offsets.h>
 
-/* We need to dummy out DT_NODE_HAS_STATUS when building the unittests.
- * Including devicetree.h would require generating dummy header files
- * to match what gen_defines creates, so it's easier to just dummy out
- * DT_NODE_HAS_STATUS.
- */
-#ifdef ZTEST_UNITTEST
-#define DT_NODE_HAS_STATUS(node, status) 0
```
Comment (tejlmand) [inline include/zephyr/linker/linker-defs.h:34]: Curious, where is DT_NODE_HAS_STATUS(node, status) defined now / why is it no longer needed ?

**Review by tejlmand — COMMENTED** [review]
Looks ok to me to relocate these file.

I think it's a little more clear to users that these headers are in-tree, but empty, compared to generate empty files as proposed in #69461, but both approaches are acceptable to me.

File: include/zephyr/linker/linker-defs.h (hunk 1 above)
Comment (cfriedt) [reply include/zephyr/linker/linker-defs.h:34]: I'm also curious about that - could it possiby be documentation related?

**Comment by ycsin** [comment]
Changed this to draft - I prefer #69461

**Pull request closed without merging**